src/uint/mul/schoolbook.rs

```diff
@@ -87,7 +87,7 @@ pub const fn wrapping_mul(lhs: &[Limb], rhs: &[Limb], out: &mut [Limb]) {
 
         while k < lhs.len() {
             let j = k - i;
-            if k == rhs.len() {
+            if j == rhs.len() {
```

Comment:
Was this the only issue? I saw some other test failures besides the out-of-bounds panic

Reply:
Yep, seems to fix all the tests. Besides the panic it could cause incorrect results, some of the new tests capture that:

```
thread 'uint::mul::tests::wrapping_mul_mixed' panicked at src/uint/mul.rs:357:9:
assertion `left == right` failed
  left: Uint(0x0000000000000000751EA4E5BF0EB289)
 right: Uint(0x7527C959D947A4F1751EA4E5BF0EB289)
```